pkg/armhelpers/util.go

```diff
@@ -3,13 +3,34 @@ package armhelpers
 import (
 	"fmt"
 	"net/url"
+	"regexp"
 	"strconv"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/arm/compute"
 	log "github.com/sirupsen/logrus"
 )
 
+const (
+	// TODO: merge with the RP code
+	// here there are 4 capture groups
+	//  #1 is the orchestrator (dcos|k8s|swarm|swarmm)
+	//  #2 is the agent pool name, which can contain -s
+	//  #3 is the Cluster ID for the cluster
+	//  #4 is the index of the vm in the agent pool/master pool
+	k8sLinuxVMNamingFormat         = "^([^-]+)-(.+)-([0-9a-fA-F]{8})-{0,2}([0-9]+)$"
```

Comment:
If it's k8s we don't need to generalize the first part IMO. We already have an assumption in upgradecluster.go and all the BI.

Reply:
plz explain

Reply:
Discussed offline